packages/server/src/internals/procedure.ts

```diff
@@ -44,11 +44,12 @@ export type ProcedureResolver<TContext, TInput, TOutput> = (opts: {
   type: ProcedureType;
 }) => Promise<TOutput> | TOutput;
 
-interface ProcedureOptions<TContext, TInput, TOutput, TParsedOutput> {
-  middlewares: Array<MiddlewareFunction<any, any>>;
+interface ProcedureOptions<TContext, TInput, TOutput, TMeta, TParsedOutput> {
```

Comment:
Align arg order to `Router`

```suggestion
interface ProcedureOptions<TContext, TMeta, TInput, TOutput, TParsedOutput> {
```